Practical_Training/Common/Filters/ActionFilter.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Newtonsoft.Json;
+
+namespace Practical_Training.Common.Filters
+{
+    public class ApiActionFilter : IActionFilter
```

Comment:
It's not clear what is the purpose of this filter, can you explain this please

Reply:
used to log the request **(OnActionExecuting)** and the response **(OnActionExecuted)**  for each action in the ApiController, to prevent the repeat of this  code in every action-controller